go/interpreter/lfvm/instructions_test.go

```diff
@@ -1879,6 +1755,85 @@ func TestInstructions_ReturnDataCopy_ReturnsErrorOnParameterOverflow(t *testing.
 	}
 }
 
+func TestInstructions_MCopy_ReturnsErrorOnFailure(t *testing.T) {
+
+	tests := map[string]struct {
+		srcOffset, destOffset, size uint64
+		expectedError               error
+		gasRemoved                  uint64
+	}{
+		"does nothing if size is zero": {
+			srcOffset: 128, destOffset: 1024, size: 0,
+		},
+		"returns error when failed write memory expansion": {
+			srcOffset: math.MaxUint64, destOffset: 0, size: 1,
+			expectedError: errOverflow,
+		},
+		"returns error when failed read memory expansion": {
+			srcOffset: 0, destOffset: math.MaxUint64, size: 1,
+			expectedError: errOverflow,
+		},
```

Comment:
nit: not sure, but I think the labels are interchanged; the first one should be the read case, the second the write case.

Reply:
right